Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py

```diff
@@ -169,10 +175,30 @@ def main(cls, args, repository, why=None, redact=False, **kwargs):
             sys.stderr.write("'{}' is an invalid branch name, cannot create it\n".format(args.issue))
             return 1
 
-        remote_re = re.compile('remotes/.+/{}'.format(re.escape(args.issue)))
-        for branch in repository.branches:
-            if branch == args.issue or remote_re.match(branch):
-                sys.stderr.write("'{}' already exists\n".format(args.issue))
+        if args.issue in repository.branches_for(remote=target_remote):
+            if not args.delete_existing:
+                sys.stderr.write("'{}' exists on the remote '{}' and will be overwritten by a push\n".format(args.issue, target_remote))
+                if args.delete_existing is False:
+                    return 1
+
+        for remote in repository.source_remotes():
+            if args.issue in repository.branches_for(remote=remote):
+                sys.stderr.write("'{}' cannot be used as a PR branch, it exists on remote '{}'\n".format(args.issue, remote))
+                return 1
+        if args.issue in repository.branches_for(remote=False):
+            if args.delete_existing:
+                log.info("Locally deleting existing branch '{}'".format(args.issue))
+                if run([repository.executable(), 'branch', '-D', args.issue], cwd=repository.root_path).returncode:
+                    sys.stderr.write("Failed to locally delete '{}'\n".format(args.issue))
+            elif args.delete_existing is None:
+                log.warning("Rebasing existing branch '{}' instead of creating a new one".format(args.issue))
+                if run([repository.executable(), 'rebase', 'HEAD', args.issue, '--autostash'], cwd=repository.root_path).returncode:
+                    sys.stderr.write("Failed to rebase '{}'\n".format(args.issue))
+                    return 1
```

Comment:
If the rebase introduces conflicts, the rebase hasn't _failed_, it just needs manual fixups and a call to `git rebase --continue`. Can we avoid writing our own error message and count on git's output to be informative enough?
```suggestion
                if run([repository.executable(), 'rebase', 'HEAD', args.issue, '--autostash'], cwd=repository.root_path).returncode:
                    return 1
```

Saying "Failed to rebase" when there are conflicts and the original commits haven't been applied yet may make it seem like we have caused data loss, when in reality the local changes will come back after running `git rebase --continue`, and the original branch can be restored with `git rebase --abort`.

Reply:
Will remove this error message, was trying to be helpful, but yeah, I can see how our editorializing here is not useful.